### PR TITLE
Add auto `test` debug mode for launch.json (#1780)

### PR DESCRIPTION
**Pull request by qzyse2017**
I tested it manually and it worked correctly for #1780 , I don' t know how to add a unit test myself, I run the tests already in the repo and it performed just as before I made a change.

### CODE REVIEW

File: src/goDebugConfiguration.ts [hunk 1]
```diff
@@ -29,16 +29,22 @@ export class GoDebugConfigurationProvider implements vscode.DebugConfigurationPr
...
+			} else {
+				debugMode = 'debug';
+			}
 
 			debugConfiguration = {
 				'name': 'Launch',
 				'type': 'go',
 				'request': 'launch',
-				'mode': 'debug',
+				'mode': debugMode,
```
Comment (ramya-rao-a) [inline src/goDebugConfiguration.ts]: When there is a debug configuration in the launch.json file, this `if` block (starting at line 27) will not be entered. So this fix will only get applied to cases when there is no debug configuration in the launch.json file.

I would suggest to update the mode after this `if` block

Comment (ramya-rao-a) [inline src/goDebugConfiguration.ts]: When there is a debug configuration in the launch.json file, this `if` block (starting at line 27) will not be entered. So this fix will only get applied to cases when there is no debug configuration in the launch.json file.

I would suggest to update the mode after this `if` block

File: src/goDebugConfiguration.ts [hunk 2]
```diff
@@ -29,16 +29,22 @@ export class GoDebugConfigurationProvider implements vscode.DebugConfigurationPr
 			if (!activeEditor || activeEditor.document.languageId !== 'go') {
 				return;
 			}
+			let debugMode;
+			let testFileRegExp = RegExp('.*_test\.go$', 'g');
+			if (testFileRegExp.test(activeEditor.document.fileName) === true) {
```
Comment (ramya-rao-a) [inline src/goDebugConfiguration.ts]: isnt it simpler with `if (activeEditor.document.fileName.endsWith('_test.go'))` ?

**Comment by qzyse2017** [comment]
sorry I had deleted the branch which I used to change the code before you reviewed😥.
It seems there is no way I can continue to commit to this pull request.
Created a new pull request [here](https://github.com/Microsoft/vscode-go/pull/1832)

**Comment by ramya-rao-a** [comment]
Fixes #1780

**Pull request closed without merging**